File: 666B.cpp
#include <bits/stdc++.h>

using namespace std;
using ll = long long;
int mx = 1e5 + 1;

int solve(int a1, int a2, int v, int b)
{
    int on = a1, tw = a2;
    int ans1 = 1, ans2 = 1;
    int b2 = b;
    while (b > 0)
    {
        if (b & 1)
            ans1 *= a1;
        a1 *= a1;
        b >>= 1;
    }
    while (b2 > 0)
    {
        if (b2 & 1)
            ans2 *= a2;
        a2 *= a2;
        b2 >>= 1;
    }
    if (abs(a1 - v) > abs(a2 - v))
        return on;
    else
        return tw;
}

int main()
{
    int n;
    ll ans = 0;
    cin >> n;
    vector<int> v(n);
    for (auto a : v)
        cin >> a;
    sort(v.begin(), v.end());
    double t = 1 / (double)(n - 1);
    double a = pow(v[n - 1], t);
    int val = solve(a, a + 1, v[n - 1], n - 1);
    cout << setprecision(10) << val << " " << a << " " << t << '\n';
    int k = val;
    for (int i = 0; i < n; i++)
    {
        if (i == 0)
        {
            ans += (v[i] - 1);
        }
        else
        {
            ans += abs(v[i] - val);
            val *= t;
        }
    }

    cout << ans << '\n';
}

File: 666B.py
n = int(input())
arr = list(map(int, input().split()))
arr.sort()
l = 0
r = 100000
val = 1000000007
v = 0
while(l <= r):
    m = int((r-l)/2)
    print(str(l) + " " + str(r) + " " + str(m))
    k = m ** (n-1)
    q = abs(arr[n-1]-k)
    if (q < val):
        val = q
        v = m
    if(k > arr[n-1]):
        r = m
    elif(k < arr[n-1]):
        l = m
t = v
ans = 0
for i in range(n):
    if i == 0:
        ans += (arr[i] - 1)
    else:
        ans += (arr[i] - v)
        v *= t
print(ans)
